bin/benchmark

```diff
@@ -8,52 +8,59 @@ NCORES=$(grep ^cpu\\scores /proc/cpuinfo | uniq |  awk '{print $4}')
 
 # benchmark pytest-benchmark based scripts
 benchmark_pytest() {
-    source activate ./env
+    #source activate ./env
     # set threads to one
+    echo "Benchmark pytest: $1 ($(date +"%Y-%m-%d-%H-%M-%S"))"
     export OMP_NUM_THREADS=1
     export MKL_NUM_THREADS=1
     export MKL_DOMAIN_NUM_THREADS=1
-
+    
+    mkdir -p "$BENCHMARK_LOG_PATH" "$BENCHMARK_DATA_PATH"
     pytest "$1/benchmarks.py" --benchmark-storage="file://$BENCHMARK_DATA_PATH" \
-        --benchmark-save=$1 --benchmark-sort=name --benchmark-min-rounds=5\
-        >$BENCHMARK_LOG_PATH/$1.out 2> $BENCHMARK_LOG_PATH$1.err
+        --benchmark-save=$1 --benchmark-sort=name --benchmark-min-rounds=5 \
+        > "$BENCHMARK_LOG_PATH/$1.out" 2> "$BENCHMARK_LOG_PATH/$1.err"
 }
 
 # benchmark Julia BenchmarkTools based scripts
 benchmark_julia() {
+    echo "Bechnmark julia: yao ($(date +"%Y-%m-%d-%H-%M-%S"))"
     export JULIA_NUM_THREADS=1
     export OMP_NUM_THREADS=1
     export MKL_NUM_THREADS=1
 
+    mkdir -p "$BENCHMARK_LOG_PATH" "$BENCHMARK_DATA_PATH"
     julia --project yao/benchmarks.jl\
         $BENCHMARK_DATA_PATH/yao.csv\
         $BENCHMARK_DATA_PATH/yao_qcbm.csv\
         $BENCHMARK_DATA_PATH/yao_qcbm_batch.csv\
-        > $BENCHMARK_LOG_PATH/yao.out 2> $BENCHMARK_LOG_PATH/yao.err
+        > "$BENCHMARK_LOG_PATH/yao.out" 2> "$BENCHMARK_LOG_PATH/yao.err"
+}
+
+# benchmark misc
+benchmark_misc() {
+    echo "Benchmark misc: $1 ($(date +"%Y-%m-%d-%H-%M-%S"))"
+    mkdir -p "$BENCHMARK_LOG_PATH" "$BENCHMARK_DATA_PATH"
+    "$1/benchmarks.sh" "$BENCHMARK_DATA_PATH" > "$BENCHMARK_LOG_PATH/$1.out" 2> "$BENCHMARK_LOG_PATH/$1.err"
+
 }
 
 install() {
     # install Python dependencies
     echo "creating local virtualenv"
-    conda create -y --prefix=env
+    conda create -y --prefix=env python=3.8
     echo "activating virtualenv"
-    source activate ./env
+    conda activate ./env
```

Comment:
I think conda activate won't always work unless it is setup in one's bash file?

Reply:
`source` is an older other name for conda (I think pre 4.8). The shebang says this script is using `sh`, which doesn't have a  `source` buildin. `conda` has to be on the PATH anyway and I think this might be more intuitive.